client/packages/common/src/ui/components/inputs/Filters/FilterMenu.tsx

```diff
@@ -130,6 +177,16 @@ const getFilterComponent = (
           remove={() => removeFilter(filter)}
         />
       );
+    case 'date':
+    case 'dateTime':
+      return (
+        <DateFilter
+          key={`${filter.urlParameter}${
+            filter.range ? '_' + filter.range : ''
+          }`}
```

Comment:
If it's part of a range group, append the range value (from/to) here to ensure keys are unique (they have the same urlParameter)